src/test/java/com/fasterxml/jackson/core/json/async/AsyncNonStandardNumberParsingTest.java

```diff
@@ -144,8 +145,57 @@ public void testLeadingPlusSignInDecimalDefaultFail() throws Exception {
             p.nextToken();
             fail("Expected exception");
         } catch (Exception e) {
-            //the message does not match non-async parsers
-            verifyException(e, "Unrecognized token '+123'");
+            verifyException(e, "Unexpected character ('+'");
+        } finally {
+            p.close();
+        }
+    }
+
+    public void testLeadingPlusSignInDecimalDefaultFail2() throws Exception {
+        final String JSON = "[ +0.123 ]";
+
+        // without enabling, should get an exception
+        AsyncReaderWrapper p = createParser(DEFAULT_F, JSON, 1);
+        assertToken(JsonToken.START_ARRAY, p.nextToken());
+        try {
+            p.nextToken();
+            fail("Expected exception");
+        } catch (Exception e) {
```

Comment:
Should probably catch more specific exception (`StreamReadException`?), but since all other tests for some reason also use `Exception` it's fine here too I suppose.

Reply:
changed